example/ring/ring.cpp

```diff
@@ -106,25 +113,37 @@ class ring_recipe: public arb::recipe {
 };
 
 struct cell_stats {
-    using size_type = std::uint64_t;
+    using size_type = unsigned;
     size_type ncells = 0;
     size_type nsegs = 0;
     size_type ncomp = 0;
 
-    cell_stats(arb::distributed_context_handle ctx, arb::recipe& r) {
-        size_type nranks = ctx->size();
-        size_type rank = ctx->id();
+    cell_stats(arb::recipe& r) {
+#ifdef ARB_MPI_ENABLED
+        int nranks, rank;
+        MPI_Comm_rank(MPI_COMM_WORLD, &rank);
```

Comment:
Yes, but I just wanted to keep it simple.
It is fair to assume that things will have already broken if MPI_COMM_WORLD is invalid, because the `with_mpi` guard (which does perform error checking) will have thrown an exception.

I just keep it simple, because this is "pedagogical" code.